src/synthesiser/Synthesiser.cpp

```diff
@@ -197,6 +197,70 @@ void Synthesiser::emitCode(std::ostream& out, const RamStatement& stmt) {
             };
         }
 
+        std::pair<std::stringstream, std::stringstream> getPaddedRangeBounds(const RamRelation& rel,
+                const std::vector<RamExpression*>& rangePatternLower,
+                const std::vector<RamExpression*>& rangePatternUpper) {
+            std::stringstream low;
+            std::stringstream high;
+
+            size_t realArity = rel.getArity();  // making this distinction for provenance
```

Comment:
Where possible, such as here, do not use inline comments. Put it on the line before.